homeassistant/components/blebox/__init__.py

```diff
@@ -66,17 +67,27 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 @callback
 def create_blebox_entities(
-    hass, config_entry, async_add_entities, entity_klass, entity_type
+    hass,
+    config_entry,
+    async_add_entities,
+    entity_klass,
+    entity_type,
+    entity_descriptors=None,
 ):
     """Create entities from a BleBox product's features."""
 
-    product = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
+    product: Box = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
     entities = []
 
     if entity_type in product.features:
         for feature in product.features[entity_type]:
-            entities.append(entity_klass(feature))
-
+            if entity_descriptors:
+                for description in entity_descriptors:
+                    if description.key == feature.device_class:
+                        entities.append(entity_klass(feature, description))
+                        break
+            else:
+                entities.append(entity_klass(feature))
```

Comment:
You don't need to adjust `create_blebox_entities` in this PR.
```suggestion
            entities.append(entity_klass(feature))

```